### DISCUSSION TITLE
Internal Exception error when sending playerpacket via "sendPacketToViewersAndSelf()"

### DISCUSSION THREAD

**Discussion by Calc-oholic** [Minestom/Minestom#1910] [Help] [answered]
Hey! I seem to get this error when attempting to send ParticleCreator particle packets to the player. The error occurs right after I run the "/lightspeed" command, and kicks me with this error. (Code below)

![image](https://github.com/Minestom/Minestom/assets/104602895/fd2e983d-8871-493c-b937-2382ce0900ff)

**CODE:**

`public class lightspeed extends Command {

    public lightspeed() {
        super("lightspeed");
        setDefaultExecutor((sender, context) -> {
            sender.sendMessage("Usage:  /lightspeed");
        });

        addSyntax((sender, context) -> {
            if (!sender.isPlayer()) {
                sender.sendMessage("Only players can use this command.");
                return;
            }
            Entity player = sender.asPlayer();
            createCustomDimension(player);
            System.out.println("Player ran lightspeed command");
        });
    }

    private void createCustomDimension(Entity player) {
        InstanceManager instanceManager = MinecraftServer.getInstanceManager();
        InstanceContainer customInstance = instanceManager.createInstanceContainer();
        MinecraftServer.getInstanceManager().registerInstance(customInstance);
        customInstance.setGenerator(GenerationUnit::modifier);

        Pos teleportPosition = new Pos(0, 100, 0);
        player.setInstance(customInstance);
        player.teleport(teleportPosition);

        MinecraftServer.getSchedulerManager().buildTask(() -> spawnStarsAroundPlayer(player))
                .delay(20, TimeUnit.SERVER_TICK)
                .repeat(1, TimeUnit.SERVER_TICK)
                .schedule();
    }

    private void spawnStarsAroundPlayer(Entity player) {
        Random random = new Random();
        Pos playerPosition = player.getPosition();

        int numStars = 100;

        for (int i = 0; i < numStars; i++) {
            double distance = 50.0;
            double angle = random.nextDouble() * Math.PI * 2.0;
            double x = playerPosition.blockX() + distance * Math.cos(angle);
            double y = playerPosition.blockY() + random.nextDouble() * 10.0 - 5.0;
            double z = playerPosition.blockZ() + distance * Math.sin(angle);

            ParticlePacket particle = ParticleCreator.createParticlePacket(Particle.DUST, x, y, z, 0f, 0f, 0f, 1);

            player.sendPacketToViewersAndSelf(particle);
        }
    }
}`

Any help would be much appreciated, thanks.

**Comment by KrystilizeNevaDies** [comment, marked as answer]
Some particle types require additional information to be written. see here: https://wiki.vg/Protocol#Particle
We can see that dust requires four floats to be written: 
![image](https://github.com/Minestom/Minestom/assets/57762380/b1a12540-e252-471f-81ab-045637920a5c)
so you can likely change this line `ParticlePacket particle = ParticleCreator.createParticlePacket(Particle.DUST, x, y, z, 0f, 0f, 0f, 1);`
to this:
```
ParticlePacket particle = ParticleCreator.createParticlePacket(Particle.DUST, true, x, y, z, 0f, 0f, 0f, 0, 1, writer -> {
    writer.writeFloat(0.3f);
    writer.writeFloat(0.3f);
    writer.writeFloat(0.3f);
    writer.writeFloat(0.3f);
});
```
